src/structures/suggestion.ts

```diff
@@ -0,0 +1,265 @@
+import { type Snowflake, User, type GuildTextBasedChannel, GuildMember, EmbedBuilder } from "discord.js";
+import { Votable } from "./votable.ts";
+import {
+	Suggestion as DbSuggestion,
+	type SuggestionSelect,
+	type SuggestionStatus,
+} from "../schemas/suggestion.ts";
+import db from "../db.ts";
+import { eq } from "drizzle-orm";
+import type { ConfigSelect } from "../schemas/config.ts";
+import { SuggestionUtil } from "./suggestion-util.ts";
+import { client } from "../client.ts";
+
+export const SUGGESTION_USER_ALREADY_VOTED = "UserAlreadyVoted";
+
+interface CreateSuggestionOptions {
+	title: string;
+	description?: string;
+	channel: GuildTextBasedChannel;
+	member: GuildMember;
+	dbConfig?: ConfigSelect;
+}
+
+export class Suggestion extends Votable<Snowflake> {
+	/**
+	 * The maximum length for the suggestion title.
+	 *
+	 * @default 100
+	 */
+	public static readonly MAX_TITLE_LENGTH = 100;
+
+	/**
+	 * The maximum length for the suggestion description.
+	 *
+	 * @default 2000
+	 */
+	public static readonly MAX_DESCRIPTION_LENGTH = 2000;
+
+	/**
+	 * The maximum length for the status reason.
+	 *
+	 * @default 2000
+	 */
+	public static readonly MAX_REASON_LENGTH = 2000;
+
+    public static readonly VOTE_SERIALIZE_SEPARATOR = ','
+
+	constructor(protected readonly id: number) {
+		super();
+	}
+
+	/** Upvote the suggestion. */
+	public override async upvote(userId: string, dbConfig?: ConfigSelect): Promise<void> {
+		if (await this.canVote()) return;
+
+		super.upvote(userId);
+
+		const dbSuggestion = db
+			.select({
+				downvotedBy: DbSuggestion.downvotedBy,
+				upvotedBy: DbSuggestion.upvotedBy,
+			})
+			.from(DbSuggestion)
+			.where(eq(DbSuggestion.id, this.id))
+			.get();
+
+		const upvotes = new Set(dbSuggestion?.upvotedBy);
+		const downvotes = new Set(dbSuggestion?.downvotedBy);
+
+		upvotes.add(userId);
+		downvotes.delete(userId);
+
+		const updatedSuggestion = await db
```

Comment:
You can implement voting in one query using `coalesce` with `instr` like [this](https://github.com/AsyncBanana/vote-Potomac/blob/main/src/pages/api/%5Btype%5D/vote.ts#L27). However, since this is local SQLite (the example I gave was using SQLite over HTTP), if you think the readability cost outweighs the performance benefits, stick with two separate queries. However, if you don't implement this in one query, you should implement this logic in a transaction